backend/dal/Services/Concrete/ParcelService.cs

```diff
@@ -38,14 +39,20 @@ public class ParcelService : BaseService<Parcel>, IParcelService
         /// <returns></returns>
         public IEnumerable<Parcel> GetNoTracking(double? neLat = null, double? neLong = null, double? swLat = null, double? swLong = null, int? agencyId = null, int? propertyClassificationId = null)
         {
+            // Check if user has the ability to view sensitive properties.
+            var userAgencies = this.User.GetAgencies();
+            var viewSensitive = this.User.HasRole(Security.RoleClaim.SensitiveView);
+
             IQueryable<Parcel> query = null;
             if (neLat != null && neLong != null && swLat != null && swLong != null)
             {
-                query = this.Context.Parcels.AsNoTracking().Where(parcel =>
-                    parcel.Latitude <= neLat &&
-                    parcel.Latitude >= swLat &&
-                    parcel.Longitude <= neLong &&
-                    parcel.Longitude >= swLong);
+                // Users may only view sensitive properties if they have the `sensitive-view` claim and belong to the owning agency.
+                query = this.Context.Parcels.AsNoTracking().Where(p =>
+                    (!p.IsSensitive || (viewSensitive && userAgencies.Contains(p.AgencyId))) &&
```

Comment:
Does this support the concept of sub-agencies? Ie, would this allow a user from the ministry of health to see a sensitive island health property?

Reply:
Yes, we will include a claim that is an array (sub-agencies) called `agencies`.  Which we can use `User.GetAgencies()` to extract.